src/PresentationalComponents/ExecutiveReport/Download.js

```diff
@@ -23,30 +24,29 @@ const DownloadExecReport = ({ isDisabled }) => {
 
   const dataFetch = async () => {
     setLoading(true);
-    const { BuildExecReport } = await import(
-      /* webpackChunkName: 'BuildExecReport' */ './BuildExecReport'
-    );
     dispatch(addNotification(exportNotifications.pending));
 
     try {
-      const [statsSystems, statsReports, topActiveRec] = await Promise.all([
-        (await Get(STATS_SYSTEMS_FETCH_URL)).data,
-        (await Get(STATS_REPORTS_FETCH_URL)).data,
-        (
-          await Get(
+      const [statsSystems, statsReports, topActiveRec] = (
+        await Promise.all([
+          Get(STATS_SYSTEMS_FETCH_URL),
+          Get(STATS_REPORTS_FETCH_URL),
+          Get(
             RULES_FETCH_URL,
             {},
             { limit: 3, sort: '-total_risk,-impacted_count', impacting: true }
-          )
-        ).data,
-      ]);
+          ),
+        ])
+      ).map(({ data }) => data);
 
-      const report = BuildExecReport({
-        statsReports,
-        statsSystems,
-        topActiveRec,
-        intl,
-      });
+      const report = (
+        <BuildExecReport
```

Comment:
I believe that the report should not be pre-rendered when it is passed to the generator service. For example, [vulnerability](https://github.com/RedHatInsights/vulnerability-ui/blob/5e9c2cce07773dc95fea302bcd8b94b9a08233ea/src/Components/SmartComponents/Reports/DownloadSystemsReport.js#L85) also does not return pre-rendered component.

Reply:
What do you mean with "pre-rendered"?

Reply:
If you mean call the `BuildExecReport` component (properly) as a component then, no it should not be called as a function. Yes, it works and Vulnerability is doing it too this way, but it is not correct and can interfere with Reacts correct behaviour. 

If it returns JSX it is a component, and components should be called as such.

Reply:
ah, I got confused as I was looking at vulnerability while reviewing